### PR TITLE
Update GPUTreeShap

### PR DESCRIPTION
**Pull request by RAMitchell**
Fixes integer overflow

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -50,6 +49,7 @@ if (USE_CUDA)
     CUDA_STANDARD 14
     CUDA_STANDARD_REQUIRED ON
     CUDA_SEPARABLE_COMPILATION OFF)
+  target_link_libraries(objxgboost PRIVATE GPUTreeShap)
```
Comment (hcho3) [inline src/CMakeLists.txt:52]:
```suggestion
  target_link_libraries(objxgboost PRIVATE GPUTreeShap::GPUTreeShap)
```
for aesthetics.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -50,6 +49,7 @@ if (USE_CUDA)
     CUDA_STANDARD 14
     CUDA_STANDARD_REQUIRED ON
     CUDA_SEPARABLE_COMPILATION OFF)
+  target_link_libraries(objxgboost PRIVATE GPUTreeShap::GPUTreeShap)
```
Comment (hcho3) [inline src/CMakeLists.txt:52]:
```suggestion
  target_link_libraries(objxgboost PUBLIC GPUTreeShap::GPUTreeShap)
```

Comment (hcho3) [reply src/CMakeLists.txt:52]: This ensures that the GPU shap header is accessible to the gtest binary.

**Review by trivialfis — APPROVED** [review]

File: src/CMakeLists.txt (hunk 2 above)
Comment (RAMitchell) [reply src/CMakeLists.txt:52]: I thought it should be private based on this: https://cmake.org/pipermail/cmake/2016-May/063400.html

Comment (hcho3) [reply src/CMakeLists.txt:52]: Never mind. tests/cpp/predictor/test_gpu_predictor.cu doesn't directly access `GPUTreeShap/gpu_treeshap.h`

**Pull request merged**